controllers/mentor.go

```diff
@@ -105,3 +131,89 @@ func FetchAllMentors(w http.ResponseWriter, r *http.Request) {
 
 	utils.RespondWithJson(r, w, mentors)
 }
+
+// /mentor/dashboard/ functions
+
+func CreateMentorDashboard(mentor models.Mentor, db *gorm.DB) MentorDashboard {
+	var projects []models.Project
+	var projectsInfo []ProjectInfo
+	var students []StudentInfo
+
+	db.Table("projects").
+		Where("mentor_id = ? OR secondary_mentor_id = ?", mentor.ID, mentor.ID).
+		Select("name", "repo_link", "commit_count", "pull_count", "lines_added", "lines_removed").
+		Find(&projects)
+
+	for _, project := range projects {
+		projectInfo := ProjectInfo{
+			Name:     project.Name,
+			RepoLink: project.RepoLink,
+
+			// stats table
+			CommitCount:  project.CommitCount,
+			PullCount:    project.PullCount,
+			LinesAdded:   project.LinesAdded,
+			LinesRemoved: project.LinesRemoved,
+		}
+		projectsInfo = append(projectsInfo, projectInfo)
+
+		var modelStudent models.Student
+		for _, studentUsername := range strings.Split(project.Contributors, ",") {
+			db.Table("students").Where("username = ?", studentUsername).
+				Select("name", "username").First(&modelStudent)
```

Comment:
You can instantiate a `StudentInfo` struct and directly provide it to `.First()`. Instantiating a `models.Student` is unnecessary here.